autoscheduler/scraper/tests/api_tests.py

```diff
@@ -0,0 +1,267 @@
+from scraper.models.instructor import Instructor
+from scraper.models.section import Section, Meeting
+from scraper.serializers import CourseSerializer, SectionSerializer
+
+class APITests(APITestCase):
+    """ Tests API functionality """
+    def setUp(self):
+        self.client = APIClient()
+        self.courses = [
+            Course(id='123123', dept='CSCE', course_num='181',
+                   title='Introduction to Computing', term='201931', credit_hours=3),
+            Course(id='123124', dept='CSCE', course_num='315',
+                   title='Programming Studio', term='201931', credit_hours=3),
+            Course(id='123125', dept='COMM', course_num='203',
+                   title='Public Speaking', term='201831', credit_hours=3),
+            Course(id='123126', dept='COMM', course_num='203',
+                   title='Public Speaking', term='201931', credit_hours=3),
+            Course(id='123127', dept='LAW', course_num='7500S',
+                   title='Sports Law', term='202031', credit_hours=None),
+        ]
+        test_instructors = [
+            Instructor(id='Akash Tyagi'),
+            Instructor(id='John Moore'),
+        ]
+        for instructor in test_instructors:
+            instructor.save()
+        self.sections = [
+            Section(crn=12345, id='000001', subject='CSCE', course_num='310',
+                    section_num='501', term_code='201931', min_credits='3',
+                    honors=False, web=False, max_enrollment=50,
+                    current_enrollment=40, instructor=test_instructors[0]),
+            Section(crn=12346, id='000002', subject='CSCE', course_num='310',
+                    section_num='502', term_code='201931', min_credits='3',
+                    honors=False, web=False, max_enrollment=50,
+                    current_enrollment=40, instructor=test_instructors[1]),
+        ]
+        self.meetings = [
+            Meeting(id='0000010', meeting_days=[True] * 7, start_time=time(11, 30),
+                    end_time=time(12, 20), meeting_type='LEC', section=self.sections[0]),
+            Meeting(id='0000011', meeting_days=[True] * 7, start_time=time(9, 10),
+                    end_time=time(10), meeting_type='LEC', section=self.sections[0]),
+            Meeting(id='0000020', meeting_days=[True] * 7, start_time=time(11, 30),
+                    end_time=time(12, 20), meeting_type='LEC', section=self.sections[1]),
+            Meeting(id='0000021', meeting_days=[False] * 7, start_time=time(9, 10),
+                    end_time=time(10), meeting_type='LAB', section=self.sections[1]),
+        ]
+        for course in self.courses:
+            course.save()
+        for section in self.sections:
+            section.save()
+        for meeting in self.meetings:
+            meeting.save()
+
+    def test_api_terms_displays_all_terms(self):
+        """ Tests that /api/terms returns ordered list of all terms in database """
```

Comment:
So I was just thinking that `expected` should be in order, but since it's a dictionary realized that there isn't any order.

As such, I don't think it matters the order they're returned in, as the frontend can just sort them by the keys in descending order. Thus, the docstring should change to 

```suggestion
        """ Tests that /api/terms returns a list of all terms in database """
```

Reply:
Since python 3.6, dictionaries actually do keep insertion order:
https://stackoverflow.com/questions/1867861/how-to-keep-keys-values-in-same-order-as-declared
So as long as we make a dict using a sorted list of terms, the keys will stay sorted, which we'll want to do.